Unable to build project with 0.41.0 or 0.41.1: error[E0277] with group_join_inner in polars-ops

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example


```toml
[package]
name = "test_0410"
description = "A Polars test"
authors = ["gpr"]
repository = ""
keywords = ["polars"]
license = "MIT OR Apache-2.0"
version = "0.1.0"
edition = "2021"


[dependencies]
polars = {git = "https://github.com/pola-rs/polars/", features = ["lazy"]}
```

```bash
cargo build --release
```



### Log output

```shell
/Users/guillaumepressiat/Documents/GitHub/test_0.41.0 
% cargo build --release  
   Compiling version_check v0.9.4
   Compiling proc-macro2 v1.0.86
   Compiling unicode-ident v1.0.12
   Compiling autocfg v1.3.0
   Compiling libc v0.2.155
   Compiling cfg-if v1.0.0
   Compiling crossbeam-utils v0.8.20
   Compiling libm v0.2.8
   Compiling cc v1.0.99
   Compiling serde v1.0.203
   Compiling target-features v0.1.6
   Compiling rayon-core v1.12.1
   Compiling ahash v0.8.11
   Compiling num-traits v0.2.19
   Compiling memchr v2.7.4
   Compiling once_cell v1.19.0
   Compiling zerocopy v0.7.34
   Compiling aho-corasick v1.1.3
   Compiling either v1.12.0
   Compiling quote v1.0.36
   Compiling bitflags v2.5.0
   Compiling syn v2.0.67
   Compiling getrandom v0.2.15
   Compiling crossbeam-epoch v0.9.18
   Compiling psm v0.1.21
   Compiling stacker v0.1.15
   Compiling array-init-cursor v0.2.0
   Compiling syn v1.0.109
   Compiling thiserror v1.0.61
   Compiling crossbeam-deque v0.8.5
   Compiling regex-syntax v0.8.4
   Compiling planus v0.3.1
   Compiling smartstring v1.0.1
   Compiling core-foundation-sys v0.8.6
   Compiling allocator-api2 v0.2.18
   Compiling rayon v1.10.0
   Compiling iana-time-zone v0.1.60
   Compiling polars-utils v0.41.0 (https://github.com/pola-rs/polars/#4fd59299)
   Compiling static_assertions v1.1.0
   Compiling equivalent v1.0.1
   Compiling simdutf8 v0.1.4
   Compiling chrono v0.4.38
   Compiling raw-cpuid v11.0.2
   Compiling lock_api v0.4.12
   Compiling polars-arrow v0.41.0 (https://github.com/pola-rs/polars/#4fd59299)
   Compiling parking_lot_core v0.9.10
   Compiling foreign_vec v0.1.0
   Compiling ryu v1.0.18
   Compiling regex-automata v0.4.7
   Compiling rustversion v1.0.17
   Compiling itoa v1.0.11
   Compiling ethnum v1.5.0
   Compiling strength_reduce v0.2.4
   Compiling atoi_simd v0.15.6
   Compiling smallvec v1.13.2
   Compiling streaming-iterator v0.1.9
   Compiling fast-float v0.2.0
   Compiling dyn-clone v1.0.17
   Compiling scopeguard v1.2.0
   Compiling rand_core v0.6.4
   Compiling polars-compute v0.41.0 (https://github.com/pola-rs/polars/#4fd59299)
   Compiling ppv-lite86 v0.2.17
   Compiling parking_lot v0.12.3
   Compiling regex v1.10.5
   Compiling rand_chacha v0.3.1
   Compiling heck v0.5.0
   Compiling serde_derive v1.0.203
   Compiling thiserror-impl v1.0.61
   Compiling bytemuck_derive v1.7.0
   Compiling strum_macros v0.26.4
   Compiling rand v0.8.5
   Compiling crossterm v0.27.0
   Compiling polars-core v0.41.0 (https://github.com/pola-rs/polars/#4fd59299)
   Compiling unicode-width v0.1.13
   Compiling multiversion-macros v0.7.4
   Compiling strum v0.26.2
   Compiling rand_distr v0.4.3
   Compiling bytemuck v1.16.1
   Compiling polars-ops v0.41.0 (https://github.com/pola-rs/polars/#4fd59299)
   Compiling xxhash-rust v0.8.10
   Compiling argminmax v0.6.2
   Compiling now v0.1.3
   Compiling atoi v2.0.0
   Compiling recursive-proc-macro-impl v0.1.1
   Compiling memmap2 v0.7.1
   Compiling polars-plan v0.41.0 (https://github.com/pola-rs/polars/#4fd59299)
   Compiling bytes v1.6.0
   Compiling home v0.5.9
   Compiling percent-encoding v2.3.1
   Compiling recursive v0.1.1
   Compiling polars-lazy v0.41.0 (https://github.com/pola-rs/polars/#4fd59299)
   Compiling fallible-streaming-iterator v0.1.9
   Compiling polars v0.41.0 (https://github.com/pola-rs/polars/#4fd59299)
   Compiling comfy-table v7.1.1
   Compiling streaming-decompression v0.1.2
   Compiling glob v0.3.1
   Compiling seq-macro v0.3.5
   Compiling multiversion v0.7.4
   Compiling base64 v0.22.1
   Compiling parquet-format-safe v0.2.4
   Compiling polars-arrow-format v0.1.0
   Compiling hashbrown v0.14.5
   Compiling indexmap v2.2.6
   Compiling polars-error v0.41.0 (https://github.com/pola-rs/polars/#4fd59299)
   Compiling polars-row v0.41.0 (https://github.com/pola-rs/polars/#4fd59299)
   Compiling polars-parquet v0.41.0 (https://github.com/pola-rs/polars/#4fd59299)
error[E0277]: `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>` cannot be sent between threads safely
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>` cannot be sent between threads safely
    |
    = help: the trait `for<'a> Send` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:9
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |         ^^^^ required by this bound in `group_join_inner`

error[E0277]: `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>` cannot be shared between threads safely
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>` cannot be shared between threads safely
    |
    = help: the trait `for<'a> Sync` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:16
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                ^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: std::marker::Copy` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> std::marker::Copy` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:23
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                       ^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: TotalHash` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> TotalHash` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `TotalHash`:
              bool
              char
              isize
              i8
              i16
              i32
              i64
              i128
            and 14 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:30
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                              ^^^^^^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: TotalEq` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> TotalEq` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `TotalEq`:
              bool
              char
              isize
              i8
              i16
              i32
              i64
              i128
            and 23 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:42
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                                          ^^^^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: DirtyHash` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> DirtyHash` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `DirtyHash`:
              i8
              i16
              i32
              i64
              i128
              u8
              u16
              u32
            and 7 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:52
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                                                    ^^^^^^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: IsNull` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> IsNull` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `IsNull`:
              bool
              i8
              i16
              i32
              i64
              i128
              u8
              u16
            and 15 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:64
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                                                                ^^^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: ToTotalOrd` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> ToTotalOrd` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `ToTotalOrd`:
              bool
              char
              isize
              i8
              i16
              i32
              i64
              i128
            and 14 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:217:25
    |
217 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:73
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                                                                         ^^^^^^^^^^ required by this bound in `group_join_inner`

error[E0277]: `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>` cannot be sent between threads safely
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>` cannot be sent between threads safely
    |
    = help: the trait `for<'a> Send` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:9
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |         ^^^^ required by this bound in `group_join_inner`

error[E0277]: `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>` cannot be shared between threads safely
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>` cannot be shared between threads safely
    |
    = help: the trait `for<'a> Sync` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:16
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                ^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: std::marker::Copy` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> std::marker::Copy` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:23
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                       ^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: TotalHash` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> TotalHash` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `TotalHash`:
              bool
              char
              isize
              i8
              i16
              i32
              i64
              i128
            and 14 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:30
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                              ^^^^^^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: TotalEq` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> TotalEq` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `TotalEq`:
              bool
              char
              isize
              i8
              i16
              i32
              i64
              i128
            and 23 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:42
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                                          ^^^^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: DirtyHash` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> DirtyHash` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `DirtyHash`:
              i8
              i16
              i32
              i64
              i128
              u8
              u16
              u32
            and 7 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:52
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                                                    ^^^^^^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: IsNull` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> IsNull` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `IsNull`:
              bool
              i8
              i16
              i32
              i64
              i128
              u8
              u16
            and 15 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:64
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                                                                ^^^^^^ required by this bound in `group_join_inner`

error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: ToTotalOrd` is not satisfied
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> ToTotalOrd` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `ToTotalOrd`:
              bool
              char
              isize
              i8
              i16
              i32
              i64
              i128
            and 14 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /Users/guillaumepressiat/.cargo/git/checkouts/polars-b0d90607192fd414/4fd5929/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:73
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                                                                         ^^^^^^^^^^ required by this bound in `group_join_inner`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `polars-ops` (lib) due to 16 previous errors
warning: build failed, waiting for other jobs to finish...
/Users/guillaumepressiat/Documents/GitHub/test_0.41.0 
%
```


### Issue description

I have not been able to rebuild my project with polars 0.41.0 so far. 
I've tested on an empty project and have the same error when runing cargo build 

It seems related to polars-ops and group_join_inner. 
Instead of hash_join_inner maybe?

### Expected behavior

cargo build runs fine with polars from 0.35.4 to 0.40.0

### Installed versions


I've tested both GitHub and crate versions (and run cargo update if needed).

- polars = {git = "https://github.com/pola-rs/polars/", features = ["lazy"]}
   - #e286ed3e
   - #4fd59299
- polars = {version = "0.41.0", features = ["lazy"]}
- polars = {version = "0.41.1", features = ["lazy"]}

I have a Mac M2.

In case it's helpful, I can reproduce this, also for the release 0.41.1 with rustc 1.77 in a clean project:

```
~/src $ cargo new try_polars_041
     Created binary (application) `try_polars_041` package
~/src $ cd try_polars_041 
master ~/src/try_polars_041 $ cargo add polars
    Updating crates.io index
      Adding polars v0.41.1 to dependencies.
...
    Updating crates.io index
master ~/src/try_polars_041 $ cargo build
...
error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: ToTotalOrd` is not satisfied
   --> /home/mgarcia/.cargo/registry/src/index.crates.io-6f17d22bba15001f/polars-ops-0.41.1/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> ToTotalOrd` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `ToTotalOrd`:
              bool
              char
              isize
              i8
              i16
              i32
              i64
              i128
            and 14 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /home/mgarcia/.cargo/registry/src/index.crates.io-6f17d22bba15001f/polars-ops-0.41.1/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /home/mgarcia/.cargo/registry/src/index.crates.io-6f17d22bba15001f/polars-ops-0.41.1/src/frame/join/hash_join/single_keys_dispatch.rs:322:73
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                                                                         ^^^^^^^^^^ required by this bound in `group_join_inner`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `polars-ops` (lib) due to 16 previous errors
warning: build failed, waiting for other jobs to finish...
```

Adding Polars `main` as a dependency seems to still fail. No idea why building the library works, but using it as a dependency doesn't. But let me know if I can help with anything else.

```
~/src $ cargo new polars_main
     Created binary (application) `polars_main` package

~/src $ cd polars_main 

~/src/polars_main $ cargo add --git https://github.com/pola-rs/polars.git polars
    Updating git repository `https://github.com/pola-rs/polars.git`
      Adding polars (git) to dependencies.
             Features:
             + csv
    ...
    Updating git repository `https://github.com/pola-rs/polars.git`
    Updating crates.io index

master ~/src/polars_main $ cargo run
  Downloaded strum v0.26.3
  Downloaded uuid v1.9.0
  Downloaded syn v2.0.68
  Downloaded 3 crates (319.7 KB) in 0.18s
   Compiling version_check v0.9.4
   Compiling proc-macro2 v1.0.86
...
error[E0277]: the trait bound `for<'a> <_ as polars_core::datatypes::PolarsDataType>::Physical<'a>: ToTotalOrd` is not satisfied
   --> /home/mgarcia/.cargo/git/checkouts/polars-b0d90607192fd414/cc7fc32/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ the trait `for<'a> ToTotalOrd` is not implemented for `<_ as polars_core::datatypes::PolarsDataType>::Physical<'a>`
    |
    = help: the following other types implement trait `ToTotalOrd`:
              bool
              char
              isize
              i8
              i16
              i32
              i64
              i128
            and 14 others
note: this is a known limitation of the trait solver that will be lifted in the future
   --> /home/mgarcia/.cargo/git/checkouts/polars-b0d90607192fd414/cc7fc32/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:220:25
    |
220 |                         group_join_inner(&lhs, &rhs, validate, join_nulls)
    |                         ^^^^^^^^^^^^^^^^ try adding turbofish arguments to this expression to specify the types manually, even if it's redundant
note: required by a bound in `group_join_inner`
   --> /home/mgarcia/.cargo/git/checkouts/polars-b0d90607192fd414/cc7fc32/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs:322:73
    |
312 | fn group_join_inner<T>(
    |    ---------------- required by a bound in this function
...
322 |         Send + Sync + Copy + TotalHash + TotalEq + DirtyHash + IsNull + ToTotalOrd,
    |                                                                         ^^^^^^^^^^ required by this bound in `group_join_inner`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `polars-ops` (lib) due to 16 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs
+++ b/crates/polars-ops/src/frame/join/hash_join/single_keys_dispatch.rs
@@ -63,10 +63,12 @@ pub trait SeriesJoin: SeriesSealed + Sized {
                 use BitRepr as B;
                 match (lhs, rhs) {
                     (B::Small(lhs), B::Small(rhs)) => {
-                        num_group_join_left(&lhs, &rhs, validate, join_nulls)
+                        // Turbofish: see #17137.
+                        num_group_join_left::<UInt32Type>(&lhs, &rhs, validate, join_nulls)
                     },
                     (B::Large(lhs), B::Large(rhs)) => {
-                        num_group_join_left(&lhs, &rhs, validate, join_nulls)
+                        // Turbofish: see #17137.
+                        num_group_join_left::<UInt64Type>(&lhs, &rhs, validate, join_nulls)
                     },
                     _ => {
                         polars_bail!(
@@ -139,10 +141,12 @@ pub trait SeriesJoin: SeriesSealed + Sized {
                 use BitRepr as B;
                 match (lhs, rhs) {
                     (B::Small(lhs), B::Small(rhs)) => {
-                        num_group_join_anti_semi(&lhs, &rhs, anti, join_nulls)
+                        // Turbofish: see #17137.
+                        num_group_join_anti_semi::<UInt32Type>(&lhs, &rhs, anti, join_nulls)
                     },
                     (B::Large(lhs), B::Large(rhs)) => {
-                        num_group_join_anti_semi(&lhs, &rhs, anti, join_nulls)
+                        // Turbofish: see #17137.
+                        num_group_join_anti_semi::<UInt64Type>(&lhs, &rhs, anti, join_nulls)
                     },
                     _ => {
                         polars_bail!(
@@ -214,10 +218,12 @@ pub trait SeriesJoin: SeriesSealed + Sized {
                 use BitRepr as B;
                 match (lhs, rhs) {
                     (B::Small(lhs), B::Small(rhs)) => {
-                        group_join_inner(&lhs, &rhs, validate, join_nulls)
+                        // Turbofish: see #17137.
+                        group_join_inner::<UInt32Type>(&lhs, &rhs, validate, join_nulls)
                     },
                     (B::Large(lhs), BitRepr::Large(rhs)) => {
-                        group_join_inner(&lhs, &rhs, validate, join_nulls)
+                        // Turbofish: see #17137.
+                        group_join_inner::<UInt64Type>(&lhs, &rhs, validate, join_nulls)
                     },
                     _ => {
                         polars_bail!(
@@ -279,10 +285,12 @@ pub trait SeriesJoin: SeriesSealed + Sized {
                 use BitRepr as B;
                 match (lhs, rhs) {
                     (B::Small(lhs), B::Small(rhs)) => {
-                        hash_join_outer(&lhs, &rhs, validate, join_nulls)
+                        // Turbofish: see #17137.
+                        hash_join_outer::<UInt32Type>(&lhs, &rhs, validate, join_nulls)
                     },
                     (B::Large(lhs), B::Large(rhs)) => {
-                        hash_join_outer(&lhs, &rhs, validate, join_nulls)
+                        // Turbofish: see #17137.
+                        hash_join_outer::<UInt64Type>(&lhs, &rhs, validate, join_nulls)
                     },
                     _ => {
                         polars_bail!(nyi = "Mismatch bit repr Hash Join Outer between {lhs_dtype} and {rhs_dtype}");

